File: Easy/Tournament Winner/Tournament Winner.cpp
#include <vector>
#include <unordered_map>
#include <math.h>
using namespace std;

string tournamentWinner(vector<vector<string>> competitions,
                        vector<int> results)
{
    string current_best_team = "";
    unordered_map<string, int> winners = {{current_best_team, 0}};
    for (int i = 0; i < competitions.size(); i++)
    {
        string comp_win = competitions[i][abs(results[i] - 1)];
        if (winners.find(comp_win) == winners.end())
            winners[comp_win] = 3;
        else
            winners[comp_win] += 3;

        if (winners[comp_win] > winners[current_best_team])
            current_best_team = comp_win;
    }
    return current_best_team;
}

File: Easy/Tournament Winner/Tournament Winner.py
def tournamentWinner(competitions, results):
    current_best_team = ""
    map = {current_best_team: 0}

    for i in range(len(competitions)):
        winning_team_index = 1 if results[i] == 0 else 0
        winning_team = competitions[i][winning_team_index]
        if winning_team not in map:
            map[winning_team] = 3
        else:
            map[winning_team] += 3

        if map[winning_team] > map[current_best_team]:
            current_best_team = winning_team
    return current_best_team
